_posts/2022-05-16-opentelemetry-metrics-for-data-prepper.md

```diff
@@ -0,0 +1,114 @@
+---
+layout: post
+title:  "Metrics Ingestion with Data Prepper using OpenTelemetry"
+authors:
+  - kschnitter
+  - ksternad
+date:   2022-05-06 12:00:00 -0200
+categories:
+  - technical-post
+---
+Data Prepper offers great capabilities for ingesting traces and logs into OpenSearch.
+Until now, that leaves out the third pillar of observability: Metrics.
+Data Prepper 1.4.0 introduces support for metrics received via OpenTelemetry.
+This feature aligns nicely with the OpenTelemetry traces support in Data Prepper.
+
+## Why Ingest Metrics in OpenSearch?
+
+At a first glance, OpenSearch offers great support for storing and analyzing any structured observability data.
+It provides powerful search and aggregation capabilities.
+Excellent visualizations support users in generating insights.
+The Vega visualizations lift the metrics use-case to a whole new level.
+OpenSearch already comes with alerting and anomaly detection, that nicely extends to metrics.
+
+However, OpenSearch was not built as a time series database.
+Storing metrics will be a lot less efficient with regard to resource consumption in comparison to specialized solutions.
+But, OpenSearch can shine on special use-cases: High-cardinality data sets.
+Metrics at least consists of a name, a timestamp and a value.
+Additionally, attributes can be attached that describe the nature or origin of the metrics better.
+OpenSearch does not care about the number of different values in any of those fields, specialized time series databases often do.
+Let's say you count the number of service request by a service key.
+Depending on the number of service keys, this might be problematic.
```

Comment:
> this might be problematic

This statement is a somewhat unclear: What exactly is problematic? And does the problem lie with OpenSearch or time-series databases?

Reply:
Cardinality is a problem of TSDBs not OpenSearch. I will clarify this. Should I also add some text about index roll-ups and reaggregations here to address jkowall's comment?